cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1329,6 +1337,19 @@ if(ARROW_USE_BOOST)
     endif()
     if(Boost_VERSION VERSION_GREATER_EQUAL 1.80)
```

Comment:
I don't understand, the comment above says "Boost < 1.86", why is it necessary to handle this here?